lnwallet/rpcwallet/remote_signer_client.go

```diff
@@ -0,0 +1,977 @@
+package rpcwallet
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/signrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/walletrpc"
+	"github.com/lightningnetwork/lnd/macaroons"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"gopkg.in/macaroon.v2"
+)
+
+type signerResponse = walletrpc.SignCoordinatorResponse
+
+var (
+	// ErrShuttingDown indicates that the server is in the process of
+	// gracefully exiting.
+	ErrShuttingDown = errors.New("lnd is shutting down")
+
+	// ErrRequestType is returned when the request type by the watch-only
+	// node has not been implemented by remote signer.
+	ErrRequestType = errors.New("unimplemented request by watch-only node")
+)
+
+const (
+	// defaultRetryTimeout is the default timeout used when retrying to
+	// connect to the watch-only node.
+	defaultRetryTimeout = time.Second * 1
+
+	// retryMultiplier is the multiplier used to increase the retry timeout
+	// for every retry.
+	retryMultiplier = 1.5
+
+	// defaultMaxRetryTimeout is the default max value for the
+	// maxRetryTimeout, which defines the maximum backoff period before
+	// attempting to reconnect to the watch-only node.
+	defaultMaxRetryTimeout = time.Minute * 1
+
+	// handshakeRequestID is the request ID that is reversed for the
+	// handshake with the watch-only node.
+	handshakeRequestID = uint64(1)
+)
+
+// SignCoordinatorStreamFeeder is an interface that returns a newly created
+// stream to the watch-only node. The stream is used to send and receive
+// messages between the remote signer client and the watch-only node.
+type SignCoordinatorStreamFeeder interface {
+	// GetStream returns a new stream to the watch-only node. The function
+	// also returns a cleanup function that should be called when the stream
+	// is no longer needed.
+	GetStream(streamCtx context.Context) (StreamClient, func(), error)
+
+	// Stop stops the stream feeder.
+	Stop()
+}
+
+// RemoteSignerClient is an interface that defines the methods that a remote
+// signer client should implement.
+type RemoteSignerClient interface {
+	// Start starts the remote signer client.
+	Start() error
+
+	// Stop stops the remote signer client.
+	Stop() error
+}
```

Comment:
Could you please clarify this feedback? I don't fully get how to address this.

Reply:
I specifically added the `NoOpClient`, as we specifically wanted to avoid that the `remoteSignerClient` pointer could be `nil` in `server.go`, given previous feedback from @guggero. That is what motivated exposing the `Start` & `Stop` functions in the `RemoteSignerClient` interface instead. 

Therefore I choose to implement the `MustImplementRemoteSignerClient` interface instead, so I hope that's ok :). Let me know if you still think we should set the pointer to `nil` instead!